docs_src/design/ucr/Performance-Test-Harness.md

```diff
@@ -0,0 +1,52 @@
+## Performance Test Harness
+
+## Submitters
+- Cherry Wang (IOTech)
+
+## Changelog
+<!--
+List the changes to the document, incl. state, date, and PR URL.
+
+State is one of: pending, approved, amended, deprecated
+Date is an ISO 8601 (YYYY-MM-DD) string.
+PR is the pull request that submitted the change, including information such as the diff, contributors, and reviewers.
+
+E.g.:
+- [approved](URL of PR) (2022-04-01)
+- [amended](URL of PR) (2022-05-01)
+-->
+
+## Market Segments
+- All
+
+## Motivation
+The EdgeX Performance Test aims to evaluate the efficiency of the EdgeX services in handling a large volume of data transmission from the south side to the north side in a manufacturing context. This use case is motivated by the need to understand how EdgeX performs under heavy data loads and to identify potential areas for optimization.
+
+By providing the Performance Test Harness, users can execute the Performance Tests with their own hardware and devices environment to evaluate the capability of the system.
+
+## Target Users
+- System Developer
+- System Integrator
+
+## Description
+The EdgeX Performance Test involves the south side generating a substantial amount of data and transmitting it through EdgeX services to the north side. This process aims to simulate real-world scenarios where data flow between manufacturing devices and backend systems is intensive. The observation of this data transmission allows for an in-depth analysis of EdgeX's resource utilization patterns.
+
+## Existing solutions
+N/A
```

Comment:
> How is the given use case currently implemented in the industry, with or without EdgeX?
> List and describe each approach. Highlight possible gaps.

Is this saying that people have no way to measure EdgeX performance right now? 
The purpose of this section is to highlight what people have to do to satisfy the requirement to be able to evaluate its difficulty and plan/prioritize/scope it within EdgeX.

I assume people have to write scripts to set up virtual apps and devices, produce synthetic data and ingest it into the system, then do some processing and measure the output. It is doable, but a lot of work. Within the EdgeX stack, Device Virtual, eKuiper, Record and Replay components can help in achieving the task.

Reply:
I meant there is no existing performance test harness officially provided by the community.  I listed the possible performance test approach to UCR now.